test/test_um2netcdf.py

```diff
@@ -651,6 +663,355 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+    lon_u_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    return lon_u_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_v_eg_coord():
+    # iris DimCoord imitating the real
+    # lat_v grid from CM2 (which uses the End Game grid).
+    # This grid is offset half a grid cell compared to the standard
+    # End Game latitude grid.
+    lat_v_points = np.arange(-90., 91, D_LAT_N96, dtype="float32")
+    return lat_v_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_u_eg_coord():
+    # iris DimCoord imitating the real
+    # lon_v grid from CM2 (which uses the End Game grid).
+    # This grid is offset half a grid cell compared to the standard
+    # New Dynamics longitude grid.
+    lon_u_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_u_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_nd_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lat_points = np.arange(-90, 91, D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_nd_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from ESM1.5 (which uses the New Dynamics grid).
+    lon_points = np.arange(0, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lat_standard_eg_coord():
+    # iris DimCoord imitating the standard latitude
+    # grid from CM2 (which uses the End Game grid).
+    lat_points = np.arange(-90 + 0.5*D_LAT_N96, 90.,
+                           D_LAT_N96, dtype="float32")
+    return lat_points, um2nc.LATITUDE
+
+
+@pytest.fixture
+@to_iris_dimcoord
+def lon_standard_eg_coord():
+    # iris DimCoord imitating the standard longitude
+    # grid from CM2 (which uses the End Game grid).
+    lon_points = np.arange(0.5*D_LON_N96, 360, D_LON_N96, dtype="float32")
+    return lon_points, um2nc.LONGITUDE
+
+
+class DummyCubeWithCoords(DummyCube):
+    # DummyCube with coordinates, which can be filled with
+    # iris.DimCoord objects for testing.
+    def __init__(self, dummy_cube, coords=None):
+        super().__init__(dummy_cube.item_code,
+                         dummy_cube.var_name,
+                         dummy_cube.attributes,
+                         dummy_cube.units)
+
+        # Set up coordinate dictionary to hold each dummy coordinate, with keys
+        # given by the coordinate names. This ensures that the key used to
+        # access a coordinate via the coord method matches the
+        # coordinate's name.
+        if coords is not None:
+            self.coordinate_dict = {
+                coord.name(): coord for coord in coords
+            }
+        else:
+            self.coordinate_dict = {}
+
+    def coord(self, coordinate_name):
+        return self.coordinate_dict[coordinate_name]
```

Comment:
I noticed the `coord()` lookup doesn't match the `iris` interface that raises the `CoordinateNotFoundError`. There's an example implementation here:

https://github.com/ACCESS-NRI/um2nc-standalone/blob/d8da6b1fe9ff8080f9c2f451bc28232a4316770e/test/test_um2netcdf.py#L705-L710

Once more PRs are merged, it's probably worth refactoring the multiple test cubes into 1-2 classes for reuse.

Reply:
Fixed up in 6a4c876